### PR TITLE
Filter PMP by stage. #94

### PR DESCRIPTION
**Pull request by dabreegster**

### CODE REVIEW

File: src/routes/route_check/problems_map/+page.svelte [hunk 1]
```diff
@@ -291,6 +294,20 @@
...
+
+  function show(
+    showExisting: boolean,
+    showDesign: boolean,
+    stage: "Existing" | "Design" | "",
+  ) {
+    if (!showExisting && stage == "Existing") {
+      return false;
+    }
+    if (!showDesign && stage == "Design") {
```
Comment (Pete-Y-CS) [inline src/routes/route_check/problems_map/+page.svelte:306]: I think we also need a check if it is existing and ! resolved by design
`(!showDesign && (stage == "Design" || (stage == "Existing" && !resolvedByDesign))`

**Review by Pete-Y-CS — CHANGES_REQUESTED** [review]

File: src/routes/route_check/problems_map/+page.svelte (hunk 1 above)
Comment (Pete-Y-CS) [reply src/routes/route_check/problems_map/+page.svelte:306]: Sure but my thinking is: filtering by stage it exists at.
If something exists already, and is not resolved by design then it exists in both design stage and existing stage, no?

Comment (dabreegster) [reply src/routes/route_check/problems_map/+page.svelte:306]: Ah, I see. I guess it depends what people expect the filters to mean -- "show me data I've marked as stage=design or stage=existing" vs "show me _unresolved_ issues at this stage". Could we get clarity from whoever originally asked for this? Or we could re-word these checkboxes.

**Comment by dabreegster** [comment]
Checkbox updated as discussed, and fixed the bug where the user could get into an impossible state (design stage, resolved yes). PTAL @Pete-Y-CS

**Review by Pete-Y-CS — APPROVED** [review]
lgtm!

**Pull request merged**